vdr/doc-updater.go

```diff
@@ -0,0 +1,114 @@
+package vdr
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"errors"
+
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/nuts-foundation/go-did"
+
+	nutsCrypto "github.com/nuts-foundation/nuts-node/crypto"
+)
+
+// NutsDocUpdater contains helper methods to update a nuts document.
+type NutsDocUpdater struct {
+	// keyCreator is used for getting a fresh key and use it to generate the Nuts DID
+	keyCreator nutsCrypto.KeyCreator
+}
+
+// newNamingFnForExistingDID returns a KIDNamingFunc that can be used as param in the keyCreator.New function.
+// It wraps the KIDNamingFunc with the context of the DID of the document.
+// It returns a keyID in the form of the documents DID with the new keys thumbprint as fragment.
+func newNamingFnForExistingDID(existingDID did.DID) nutsCrypto.KIDNamingFunc {
+	return func(pKey crypto.PublicKey) (string, error) {
+		ecPKey, ok := pKey.(*ecdsa.PublicKey)
```

Comment:
in general we still support multiple key types. the `jwk.New` from line 34 does these checks for you.
The two check on ecdsa keys can be removed.

Reply:
Removed them and added extra test for empty key 👍